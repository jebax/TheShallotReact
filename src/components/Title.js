import React from 'react'
import '../styles/title.css'
import { Link } from 'react-router-dom'

const Title = () => {
  return (
    <Link
      to={{ pathname: '/' }}
      id='shallotTitle'
    >
      The Shallot
    </Link>
  )
}

export default Title
