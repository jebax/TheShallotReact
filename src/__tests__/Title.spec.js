import React from 'react'
import { shallow } from 'enzyme'
import Title from '../components/Title'

describe('<Title />', () => {
  var wrapper

  beforeAll(() => {
    wrapper = shallow(<Title />)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('has a link to the home page in the title', () => {
    const link = wrapper.find('Link')
    expect(link.props().id).toEqual('shallotTitle')
    expect(link.props().to).toEqual({ pathname: '/' })
  })
})
