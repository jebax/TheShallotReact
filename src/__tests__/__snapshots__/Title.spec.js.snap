// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Title />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "The Shallot",
      "id": "shallotTitle",
      "replace": false,
      "to": Object {
        "pathname": "/",
      },
    },
    "ref": null,
    "rendered": "The Shallot",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "The Shallot",
        "id": "shallotTitle",
        "replace": false,
        "to": Object {
          "pathname": "/",
        },
      },
      "ref": null,
      "rendered": "The Shallot",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
